juju-dhx

```diff
@@ -183,11 +203,13 @@ def do_import_ids(unit, import_ids, opts):
         return
     env = get_env(opts)
     machine = unit.get('Machine')
-    annotations = env.get_annotation(machine, 'machine')
-    if annotations['Annotations'].get('import-ids') != ','.join(import_ids):
+    annotations = env.get_annotation(machine, 'machine')['Annotations'] or {}
+    import_ids = ['lp:{}'.format(iid) if ':' not in iid else iid
```

Comment:
Yeah, that's a bit of an annoyance when you have to line-break list comprehensions.  This might have been marginally clearer using `map()` instead.